dozer/cogs/filter.py

```diff
@@ -0,0 +1,266 @@
+			if results.index(filter) != (len(results)-1):
+				filter_text += "\n"
+
+		embed = discord.Embed()
+		embed.title = title="Filters for {}".format(ctx.guild.name)
+		embed.add_field(name="Filters", value=filter_text)
+		if ctx.channel.permissions_for(ctx.author).manage_guild:
+			embed.add_field(name="Commands", value="""There are several commands you can use to manage the filters.
+`{0}filter add test` - Adds test as a filter.
+`{0}filter remove 1` - Removes filter 1
+`{0}filter dm true` - Any messages containing a filtered word will be DMed
+`{0}filter whitelist` - See all of the whitelisted roles
+`{0}filter whitelist add Administrators` - Make the Administrators role whitelisted for the filter.
+`{0}filter whitelist remove Moderators` - Make the Moderators role no longer whitelisted.""".format(self.bot.command_prefix))
+		channel = await self.check_dm_filter(ctx)
+		await channel.send(embed=embed)
+		if isinstance(channel, discord.DMChannel):
+			await ctx.message.add_reaction("📬")
+	filter.example_usage = """`{prefix}filter add test` - Adds test as a filter.
+`{prefix}filter remove 1` - Removes filter 1
+`{prefix}filter dm true` - Any messages containing a filtered word will be DMed
+`{prefix}filter whitelist` - See all of the whitelisted roles
+`{prefix}filter whitelist add Administrators` - Make the Administrators role whitelisted for the filter.
+`{prefix}filter whitelist remove Moderators` - Make the Moderators role no longer whitelisted."""
+
+	@guild_only()
+	@has_permissions(manage_guild=True)
+	@filter.command()
+	async def add(self, ctx, pattern, friendly_name=None):
+		"""Add a pattern to the filter using RegEx. Any word can be added and is tested case-insensitive."""
+		new_filter = WordFilter(guild_id=ctx.guild.id, pattern=pattern, friendly_name=friendly_name or pattern)
+		with db.Session() as session:
+			session.add(new_filter)
+		embed = discord.Embed()
+		embed.title = "Filter {} added".format(new_filter.id)
+		embed.description = "A new filter with the name `{}` was added.".format(friendly_name or pattern)
+		embed.add_field(name="Pattern", value="`{}`".format(pattern))
+		await ctx.send(embed=embed)
+		self.load_filters(ctx.guild.id)
+	add.example_usage = "{prefix}filter whitelist add Swear"
+
+	@guild_only()
+	@has_permissions(manage_guild=True)
+	@filter.command()
+	async def remove(self, ctx, id):
+		"""Remove a pattern from the filter list."""
+		with db.Session() as session:
+			result = session.query(WordFilter).filter(WordFilter.id == id).one_or_none()
+			if result is None:
+				await ctx.send("Filter ID {} not found!".format(id))
+				return
+			if result.guild_id != ctx.guild.id:
+				await ctx.send("That Filter does not belong to this guild.")
+				return
+			result.enabled = False
+			await ctx.send("Filter `{}` with name `{}` deleted.".format(result.id,result.friendly_name))
+			self.load_filters(ctx.guild.id)
+	remove.example_usage = "{prefix}filter remove 7"
```

Comment:
The example usage should have the command in \`code formatting\` and include an explanation of what the command does. This also applies to the subcommands below.